lib/manageiq/session.rb

```diff
@@ -0,0 +1,65 @@
+require "patches/session_patch"
+
+module ManageIQ
+  module Session
+    FakeRequest = Struct.new(:env)
+
+    def self.store
+      @store ||= fetch_store_from_rails
+    end
+
+    def self.store=(session_store)
+      adapter_klass = "ManageIQ::Session::#{session_store.to_s.camelize}Adapter".safe_constantize
+      raise ArgumentError, "invalid session store adapter: #{session_store.inspect}" unless adapter_klass
+
+      configure_session_store(adapter_klass.new)
+
+      @store = nil
+    end
+
+    def self.configure_session_store(adapter)
+      Vmdb::Application.config.session_store(*adapter.session_store_config_args)
+      msg = "Using session_store: #{Vmdb::Application.config.session_store}"
+      $log.info("MIQ(SessionStore) #{msg}")
+      puts "** #{msg}" if !Rails.env.production? && adapter.type != :mem_cache_store
+    end
+
+    def self.revoke(session_ids)
+      store.delete_sessions(session_ids)
+    end
+
+    def self.fake_request
+      FakeRequest.new({})
+    end
+
+    def self.fetch_store_from_rails
+      middleware_store    = Rails.application
+      session_store_klass = ActionDispatch::Session::SessionObject
+
+      loop do
+        return nil              if middleware_store.nil?
+        return middleware_store if middleware_store.kind_of?(session_store_klass)
+
+        middleware_store = middleware_store.instance_variable_get(:@app)
+      end
+    end
+    private_class_method :fetch_store_from_rails
+
+    class AbstractStoreAdapter
```

Comment:
minor, but can we put this in a `lib/manageiq/sessions/abstract_store_adapter.rb` ?